### PR TITLE
Add csharp SDK

### PR DESCRIPTION
**Pull request by szunami**

### CODE REVIEW

**Comment by hpx7** [comment]
Same comment, readme update would be great

File: csharp-client-sdk/Runtime/Transport.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+namespace Hathora
+{
+    public interface Transport
+    {
+
+        public Task Connect(string stateId, string token);
+
+        public Task Write(byte[] data);
+
+        public Task<byte[]> Read();
```
Comment (hpx7) [inline csharp-client-sdk/Runtime/Transport.cs]: `Read()` makes me think this is a lower level API that reads raw bytes off the wire, but in this case I feel like `GetMessage()` or `ReadMessage()` would be more appropriate

Comment (szunami) [reply csharp-client-sdk/Runtime/Transport.cs]: I'm guessing same deal for rust api?

Comment (szunami) [reply csharp-client-sdk/Runtime/Transport.cs]: And maybe write -> writemessage?

Comment (hpx7) [reply csharp-client-sdk/Runtime/Transport.cs]: yes to both

**Pull request merged**